### PR TITLE
#159245473 Redesign remaining list pages

### PR DESCRIPTION
**Pull request by andela-fmustapha**
## What does this PR do?
It ensures all list pages look uniform in terms of design

## Description of Task to be completed?
The design used for the asset list page should be implemented on all list pages
- User list page
- Asset Types list page
- Asset Models list page
- Asset Categories list page
- Asset Sub-Categories list page
- Asset Conditions list page
- Asset Specs list page
- Incidence report page
- Allocations page

## How should this be manually tested?
 Visit all list pages and observe the uniform design
## What are the relevant pivotal tracker stories?
[#159245473](https://www.pivotaltracker.com/story/show/159245473)
## Any background context you want to add?
n/a
## Important notes
n/a
## Packages installed
n/a
## Deployment note
n/a
## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()
## Todos
- [x] Raise PR
- [x] Test

## Screenshots
screen shot 2018-08-01 at 11 56 26 pm

### CODE REVIEW

File: src/components/AssetCategoriesComponent.jsx [hunk 1]
```diff
@@ -94,21 +99,28 @@ export class AssetCategoriesComponent extends React.Component {
 
             <Table.Footer>
               <Table.Row>
-                <Table.HeaderCell colSpan="3">
+                <Table.HeaderCell colSpan="3" id="pagination-header">
```
Comment (sachungo) [inline src/components/AssetCategoriesComponent.jsx]: Same comment as previous one applies here

File: src/components/AssetCondition/AssetConditionsComponent.jsx [hunk 2]
```diff
@@ -83,26 +88,26 @@ export class AssetConditionsComponent extends React.Component {
...
-                  />
-                  }
-                  <Button
-                    circular
-                    icon="add"
-                    floated="right"
-                    data-tooltip="Add new asset condition"
-                    size="big"
-                  />
+                <Table.HeaderCell colSpan="4" id="pagination-header">
```
Comment (sachungo) [inline src/components/AssetCondition/AssetConditionsComponent.jsx]: Same previous comment applies here

File: src/components/AssetModels/AssetModelsComponent.jsx [hunk 3]
```diff
@@ -86,20 +92,27 @@ export class AssetModelsComponent extends React.Component {
 
             <Table.Footer>
               <Table.Row>
-                <Table.HeaderCell colSpan="4">
+                <Table.HeaderCell colSpan="5" id="pagination-header">
```
Comment (sachungo) [inline src/components/AssetModels/AssetModelsComponent.jsx]: Same previous comment applies here

File: src/components/AssetSpecs/AssetSpecsComponent.jsx [hunk 4]
```diff
@@ -102,20 +107,26 @@ export class AssetSpecsComponent extends React.Component {
...
-                <Table.HeaderCell colSpan="8">
-                  {!_.isEmpty(this.props.specs) &&
-                    <Pagination
-                      totalPages={this.getTotalPages()}
-                      onPageChange={this.handlePaginationChange}
-                      activePage={this.state.activePage}
-                    />
-                  }
-                  <Button circular icon="add" floated="right" data-tooltip="Add new asset spec" size="big" />
+                <Table.HeaderCell colSpan="8" id="pagination-header">
```
Comment (sachungo) [inline src/components/AssetSpecs/AssetSpecsComponent.jsx]: Same previous comment applies here

File: src/components/AssetTypesComponent.jsx [hunk 5]
```diff
@@ -79,20 +83,27 @@ export class AssetTypesComponent extends React.Component {
 
             <Table.Footer>
               <Table.Row>
-                <Table.HeaderCell colSpan="3">
+                <Table.HeaderCell colSpan="3" id="pagination-header">
```
Comment (sachungo) [inline src/components/AssetTypesComponent.jsx]: Same previous comment applies here

File: src/components/AssetsSubCategoriesComponent.jsx [hunk 6]
```diff
@@ -79,20 +84,26 @@ export class AssetSubCategoriesComponent extends React.Component {
...
-                <Table.HeaderCell colSpan="4">
-                  {!_.isEmpty(this.props.assetSubCategories) &&
-                    <Pagination
-                      totalPages={this.getTotalPages()}
-                      onPageChange={this.handlePaginationChange}
-                      activePage={this.state.activePage}
-                    />
-                  }
-                  <Button circular icon="add" floated="right" data-tooltip="Add new sub category" size="big" />
+                <Table.HeaderCell colSpan="4" id="pagination-header">
```
Comment (sachungo) [inline src/components/AssetsSubCategoriesComponent.jsx]: Same previous comment applies here

File: src/components/IncidenceReportsComponent.jsx [hunk 7]
```diff
@@ -2,12 +2,40 @@ import React from 'react';
...
-import { Container, Table, Header, Pagination } from 'semantic-ui-react';
+import { Table, Header, Pagination, Segment, Dropdown, Divider } from 'semantic-ui-react';
 
 import TableRowComponent from './TableRowComponent.jsx';
 import NavbarComponent from './NavBarComponent';
 
 import { loadIncidenceReports } from '../_actions/incidenceReports.actions';
+import '../_css/IncidenceReportsComponent.css';
+
+const rowOptions = [
```
Comment (sachungo) [inline src/components/IncidenceReportsComponent.jsx]: Please use the array created in `pageRowOptions.js`

File: src/components/IncidenceReportsComponent.jsx [hunk 8]
```diff
@@ -2,12 +2,40 @@ import React from 'react';
...
+    text: '20 Rows',
+    value: 20
+  },
+  {
+    text: '30 Rows',
+    value: 30
+  }
+];
+
+const definedPageLimits = () => (
```
Comment (sachungo) [inline src/components/IncidenceReportsComponent.jsx]: Please use the dropdown component

File: src/components/IncidenceReportsComponent.jsx [hunk 9]
```diff
@@ -70,21 +101,28 @@ export class IncidenceReportsComponent extends React.Component {
 
             <Table.Footer>
               <Table.Row>
-                <Table.HeaderCell colSpan="8">
+                <Table.HeaderCell colSpan="8" id="pagination-header">
```
Comment (sachungo) [inline src/components/IncidenceReportsComponent.jsx]: Move this after the check. I.e.
```
{!this.emptyReportsCheck() && (
  <Table.HeaderCell colSpan="8" id="pagination-header">
    <Segment.Group horizontal id="art-pagination-section">
      ...
    </Segment.Group>
  </Table.HeaderCell>
)}

File: src/components/UserDetailsComponent.jsx [hunk 10]
```diff
@@ -70,21 +63,22 @@ const UserDetailsComponent = (props) => {
         </Table.Body>
         <Table.Footer>
           <Table.Row>
-            <Table.HeaderCell colSpan="4" >
+            <Table.HeaderCell colSpan="4" id="pagination-header">
```
Comment (sachungo) [inline src/components/UserDetailsComponent.jsx]: Previous comment about "moving after the check" applies here

File: src/components/UserFeedbackComponent.jsx [hunk 11]
```diff
@@ -1,13 +1,41 @@
...
+import { Container, Header, Table, Pagination, Segment, Dropdown, Divider } from 'semantic-ui-react';
 import feedbackAction from '../_actions/userFeedback.actions';
 import NavbarComponent from './NavBarComponent';
 import TableRowComponent from './TableRowComponent';
 import LoaderComponent from './LoaderComponent';
 import ActionComponent from './ActionComponent';
 import TableHeaderComponent from '../components/common/TableHeaderComponent';
+import '../_css/UserFeedback.css';
+
+const rowOptions = [
```
Comment (sachungo) [inline src/components/UserFeedbackComponent.jsx]: Use the values defined in the option's file

File: src/components/UserFeedbackComponent.jsx [hunk 12]
```diff
@@ -1,13 +1,41 @@
...
+    text: '20 Rows',
+    value: 20
+  },
+  {
+    text: '30 Rows',
+    value: 30
+  }
+];
+
+const definedPageLimits = () => (
```
Comment (sachungo) [inline src/components/UserFeedbackComponent.jsx]: Use the dropdown component

File: src/components/UserFeedbackComponent.jsx [hunk 13]
```diff
@@ -88,16 +119,25 @@ export class UserFeedbackComponent extends React.Component {
 
           <Table.Footer>
             <Table.Row>
-              <Table.HeaderCell colSpan="5">
+              <Table.HeaderCell colSpan="5" id="pagination-header">
```
Comment (sachungo) [inline src/components/UserFeedbackComponent.jsx]: Move this after the check. I.e.
```
{this.props.hasFeedback && (
  <Table.HeaderCell colSpan="5" id="pagination-header">
    <Segment.Group horizontal id="art-pagination-section">
      ...
    </Segment.Group>
  </Table.HeaderCell>
)}

File: src/_css/LoginComponent.css [hunk 14]
```diff
@@ -89,7 +89,11 @@ input,
   padding-left: 33%; }
 
 #art-pagination-section {
-  width: 680px; }
+  width: 680px;
+  margin: 0px; }
```
[the remark this answers is not in the document]
Comment (andela-fmustapha) [reply src/_css/LoginComponent.css]: @sachungo this file is recompiled and auto generated when I edit the `sharedStyling.css` file. I don't know if I should be editing it directly. What are your thoughts?

File: src/_components/DropdownComponent.jsx [hunk 15]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Dropdown } from 'semantic-ui-react';
+import rowOptions from '../_utils/pageRowOptions';
+
+const DropdownComponent = () => (
```
[the remark this answers is not in the document]
Comment (andela-fmustapha) [reply src/_components/DropdownComponent.jsx]: Done!

[the remark this answers is not in the document]
Comment (sachungo) [reply src/_components/DropdownComponent.jsx]: @andela-fmustapha have you decided to retain this dropdown component?

File: src/_css/LoginComponent.css [hunk 16]
```diff
@@ -89,7 +89,11 @@ input,
   padding-left: 33%; }
 
 #art-pagination-section {
-  width: 680px; }
+  width: 680px;
+  margin: 0; }
```
Comment (sachungo) [inline src/_css/LoginComponent.css]: I have been informed that this is an auto-generated file from sass compiling. Ideally, we should only commit the sass file, excluding the compiled version of it. Ignore for now as I'll look into the set-up later

File: src/components/AssetModels/AssetModelsComponent.jsx (hunk 3 above)
Comment (sachungo) [reply src/components/AssetModels/AssetModelsComponent.jsx]: Please move `<Table.HeaderCell...` into `{!_.isEmpty(this.props.assetModels) && (...)` as you have done in previous component. I.e.
```
{!_.isEmpty(this.props.assetModels) && (
  <Table.HeaderCell colSpan="5" id="pagination-header">
    ...
  </Table.HeaderCell>
)}

File: src/_css/LoginComponent.css (hunk 16 above)
Comment (andela-fmustapha) [reply src/_css/LoginComponent.css]: Yes, it's auto-generated as I mentioned in my previous comment.

Comment (sachungo) [reply src/_css/LoginComponent.css]: @andela-fmustapha I have been informed that the auto-generated files should not be committed by the people who did the set-up. You can delete them and if they still show up in GitHub, use `git rm --cached <filename>` to delete them

**Pull request merged**